app/src/main/java/com/simplecity/amp_library/model/BaseFileObject.java

```diff
@@ -40,11 +40,7 @@ public boolean equals(Object o) {
 
         BaseFileObject that = (BaseFileObject) o;
 
-        if (size != that.size) return false;
-        if (fileType != that.fileType) return false;
-        if (name != null ? !name.equals(that.name) : that.name != null) return false;
-        return path != null ? path.equals(that.path) : that.path == null;
-
+        return size == that.size && fileType == that.fileType && (name != null ? name.equals(that.name) : that.name == null) && (path != null ? path.equals(that.path) : that.path == null);
```

Comment:
I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.